File: api/R/storage.R
# If storage is not loaded, it loads it
load_storage_module <- function(){
  if (!exists("storage")) {
    storage <<- reticulate::import_from_path("storage")
    return(0)
  }
  else {
    return(0)
  }
}

# Function to save sequence file in google cloud storage bucket if it doesn't exist. Returns 0 if successful
save_in_bucket <- function(
  path,
  file_extension,
  data_hash,
  directory = config::get(value = "storage")$directory,
  bucket = config::get(value = "storage")$bucket) {

  load_storage_module()
  blob_name <- file.path(directory, paste(data_hash, file_extension, sep = "."))

  blob_already_exists <- storage$check_blob_existence(bucket, blob_name)

  if (!blob_already_exists) storage$upload_blob(bucket, path, blob_name)

  return(0)
}

File: api/storage.py
from google.cloud import storage

def upload_blob(bucket_name, source_file_name, destination_blob_name):
    """Uploads a file to the bucket."""
    storage_client = storage.Client()
    bucket = storage_client.get_bucket(bucket_name)
    blob = bucket.blob(destination_blob_name)

    blob.upload_from_filename(source_file_name)

    print('File {} uploaded to {}.'.format(
        source_file_name,
        destination_blob_name))

    return(0)

def check_blob_existence(bucket_name, blob_name):
    """Checks if a blob exists in the bucket."""
    storage_client = storage.Client()
    bucket = storage_client.get_bucket(bucket_name)
    blob = bucket.blob(blob_name)

    return(blob.exists())